File: 0020-valid-parentheses/Solution2.py
# https://leetcode.com/problems/valid-parentheses/submissions/890939577/
class Solution:
    def isValid(self, s: str) -> bool:
        lazy = {"{":"}", "(":")", "[":"]"}
        queue = []


        for val in s:
            if len(queue) == 0:
                queue.append(val)
                continue
            if val in lazy.keys():
                queue.append(val)
            else:
                expected = lazy.get(queue[-1], None)
                if val == expected:
                    queue.pop() 
                else:
                    queue.append(val)

        return len(queue) == 0

File: 0507-base-7/Solution2.java
// https://leetcode.com/problems/base-7/submissions/1055843982/
class Solution {
    public String convertToBase7(int num) {
        if(num == 0){
            return "0";
        }
        return convertToBaseN(num, 7);
    }

    private String convertToBaseN(int num, int base){
        Stack<Integer> changed = new Stack<Integer>();

        int temp = Math.abs(num);
        while(temp > 0){
            changed.push(temp%base);
            temp = (int)(temp/base);
        }

        StringBuilder sb = new StringBuilder();
        if(num<0){
            sb.append("-");
        }
        while(!changed.empty()){
            sb.append(changed.pop());
        }
        return sb.toString();
    }
}
